csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyOptions.cs

```diff
@@ -0,0 +1,77 @@
+using System.Linq;
+using System.Collections.Generic;
+
+namespace Semmle.Extraction.CSharp.DependencyFetching
+{
+    /// <summary>
+    /// Dependency fetching related options.
+    /// </summary>
+    public interface IDependencyOptions
+    {
+        /// <summary>
+        /// Directories to search DLLs in.
+        /// </summary>
+        IList<string> DllDirs { get; }
+
+        /// <summary>
+        /// Files/patterns to exclude.
+        /// </summary>
+        IList<string> Excludes { get; }
+
+        /// <summary>
+        /// Whether to analyse NuGet packages.
+        /// </summary>
+        bool UseNuGet { get; }
+
+        /// <summary>
+        /// The solution file to analyse, or null if not specified.
+        /// </summary>
+        string? SolutionFile { get; }
+
+        /// <summary>
+        /// Whether to use the packaged dotnet runtime.
+        /// </summary>
+        bool UseSelfContainedDotnet { get; }
+
+        /// <summary>
+        /// Whether to search the .Net framework directory.
+        /// </summary>
+        bool ScanNetFrameworkDlls { get; }
+
+        /// <summary>
+        /// Whether to use mscorlib as a reference.
+        /// </summary>
+        bool UseMscorlib { get; }
+
+        /// <summary>
+        /// Determine whether the given path should be excluded.
+        /// </summary>
+        /// <param name="path">The path to query.</param>
+        /// <returns>True iff the path matches an exclusion.</returns>
+        bool ExcludesFile(string path);
+    }
+
+    public class DependencyOptions : IDependencyOptions
+    {
+        private static readonly DependencyOptions instance = new DependencyOptions();
```

Comment:
That is why the interface only exposes the getters to avoid this for eg. `UseNuGet` (note that the Default property returns something of type `IDependencyOptions`). However, since we are also exposing mutable data-structures like `DllDirs` and `Excludes` - you have a very valid point. I will change this.